### DISCUSSION TITLE
[Question] Combining Console and Markdown

### DISCUSSION THREAD

**Discussion by janaSunrise** [Textualize/rich#1272] [Q&A]
I have an issue. I want to combine `console.print()` and `Markdown()`, So the user in my Chat application can use `**Hello** :star:` and It would still work. I want to enable the user to use colors using `[blue] <text> [/]` and Emojis, while parsing the markdown at same time too.

Any idea how it can be achieved?

Also, How can I omit the newlines printed, when text is passed into `Markdown` and then `console.print`. 
Please guide me on how to do these. 😁

**Comment by willmcgugan** [comment]
You can't combine console markup with Markdown, unless you customize markdown.py a bit.

You could implement a basic subset of Markdown yourself, with the Text class. You will probably need a working knowledge of regexes.

> Also, How can I omit the newlines printed, when text is passed into Markdown and then console.print.

You can add `end=""` but that probably isn't what you are looking for. The Markdown class will render padded lines, so you adding a bit of text to the end won't have the desired effect.
